components/brave_rewards/browser/rewards_service_impl.h

```diff
@@ -218,9 +243,38 @@ class RewardsServiceImpl : public RewardsService,
   // Testing methods
   void SetLedgerEnvForTesting();
   void StartAutoContributeForTest();
+  void CheckInsufficientFundsForTesting();
+  void MaybeShowNotificationAddFundsForTesting(
+      base::OnceCallback<void(bool)> callback);
 
  private:
   FRIEND_TEST_ALL_PREFIXES(RewardsServiceTest, OnWalletProperties);
+  FRIEND_TEST_ALL_PREFIXES(::BraveRewardsBrowserTest,
```

Comment:
More friends than I have...

Reply:
Chromium/Google Documentation:

Testing Private Code

If you change your software's internal implementation, your tests should not break as long as the change is not observable by users. Therefore, per the black-box testing principle, most of the time you should test your code through its public interfaces.

If you still find yourself needing to test internal implementation code, consider if there's a better design. The desire to test internal implementation is often a sign that the class is doing too much. Consider extracting an implementation class, and testing it. Then use that implementation class in the original class.

Reply:
I know this may not be possible right now but we need to consider architecting our solutions so we rarely need to test private code

Reply:
I agree. The only private function I needed to access was `SetContributionAmount`. I could have done this through the UI but at the cost of making the test "flaky".